src/main/java/seedu/address/logic/commands/EditAnimalCommand.java

```diff
@@ -0,0 +1,230 @@
+            + "INDEX (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " 1"
+            + "Parameters: "
+            + NAME.getUsage() + " "
+            + DATE_OF_BIRTH.getUsage() + " "
+            + DATE_OF_ADMISSION.getUsage() + " "
+            + SPECIES.getUsage()
+            + BREED.getUsage()
+            + SEX.getUsage();
+
+    public static final String EXAMPLE_USAGE = "Example: " + COMMAND_WORD + " "
+            + "1 "
+            + NAME.getPrefix() + "ChangGoose "
+            + DATE_OF_ADMISSION.getPrefix() + "01/01/2019 "
+            + SPECIES.getPrefix() + "Dog ";
+
+    public static final String MESSAGE_SUCCESS = "Animal has been edited from: \n%s to: \n%s";
+
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided. \n%s";
+
+    public static final String MESSAGE_DO_NOT_CHANGE_ID = "You cannot change the ID of an animal. \n%s";
+    private final Index index;
+    private final EditAnimalDescriptor animalDescriptor;
+
+    /**
+     * Creates an EditCommand to edit the specified {@code Animal}
+     */
+    public EditAnimalCommand(Index index, EditAnimalDescriptor animalDescriptor) {
+        requireNonNull(index);
+        requireNonNull(animalDescriptor);
+
+        this.index = index;
+        this.animalDescriptor = new EditAnimalDescriptor(animalDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(AnimalModel model) throws CommandException {
+        requireNonNull(model);
+        List<Animal> lastShownList = model.getFilteredAnimalList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(AnimalMessages.MESSAGE_INVALID_ANIMAL_DISPLAYED_INDEX);
+        }
+
+        Animal animalToEdit = lastShownList.get(index.getZeroBased());
+        Animal editedAnimal = createEditedAnimal(animalToEdit, animalDescriptor);
+
+        model.setAnimal(animalToEdit, editedAnimal);
+        model.updateFilteredAnimalList(AnimalModel.PREDICATE_SHOW_ALL_ANIMALS);
+
+        return new CommandResult(String.format(MESSAGE_SUCCESS, AnimalMessages.format(animalToEdit),
+                AnimalMessages.format(editedAnimal)));
+    }
+
+    private static Animal createEditedAnimal(Animal animalToEdit, EditAnimalDescriptor animalDescriptor) {
+        Name updatedName = animalDescriptor.getName().orElse(animalToEdit.getName());
+        DateOfBirth updatedDateOfBirth = animalDescriptor.getDateOfBirth().orElse(animalToEdit.getDateOfBirth());
+        AdmissionDate updatedAdmissionDate = animalDescriptor.getAdmissionDate()
+                .orElse(animalToEdit.getAdmissionDate());
+        Species updatedSpecies = animalDescriptor.getSpecies().orElse(animalToEdit.getSpecies());
+        Breed updatedBreed = animalDescriptor.getBreed().orElse(animalToEdit.getBreed());
+        Sex updatedSex = animalDescriptor.getSex().orElse(animalToEdit.getSex());
+        PetId currentPetId = animalToEdit.getPetId();
```

Comment:
If copying over an animal, should the TaskList be copied over as well? I think you can consider using the `Animal` constructor with the `TaskList `as an argument

Reply:
Just copying over the `TaskList` should be fine, modifications to `TaskList `will be done using other commands.

Reply:
Yep. Good observation. Thanks!